### PR TITLE
feat: add tables, footnotes, striketroughs, and tasklists support for articles

### PR DESCRIPTION
**Pull request by patricio0312rev**
# [[articles] table and strikethrough markdown not displayed correctly](https://app.clickup.com/t/86dqhwt5a)

## Summary

- `remark-gfm` has been installed and configured as a plugin for `react-markdown`. You can check its docs [here](https://www.npmjs.com/package/remark-gfm).
- As a result, we now support not only tables and strikethroughs, but also `footnotes`, `tasklists` and `autolink literals`.
- Styles for `tasklists` have been updated to match our `checkbox input` styles.

## Steps to reproduce

1. Run the app in `local` mode.
2. Go to `/admin`portal and log in.
3. Click on the `Articles` menu item and either create or edit an existing article.
4. Any of the new supported features can be added. As an example, for tables and striketroughs, we can use:
```
## Strikethrough

~one~ or ~~two~~ tildes.

## Table

| a | b  |  c |  d  |
| - | :- | -: | :-: |
```
5. Go to `/articles` page.
6. Select the article with the new content and see the magic ✨ 

## Screenshots

image

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/js/Pages/Articles/Components/ArticleContent.tsx [hunk 1]
```diff
@@ -20,6 +21,7 @@ const externalIcon = `<svg width="12" height="12" viewBox="0 0 12 12" fill="none
 
 export const ArticleContent = ({ article }: Properties): JSX.Element => {
     const { t } = useTranslation();
+    console.log({ article });
```
Comment (alfonsobries) [inline resources/js/Pages/Articles/Components/ArticleContent.tsx:24]: 🤔

**Review by alfonsobries — CHANGES_REQUESTED** [review]
a table with three rows expands the first row

image


I would also modify some articles with different tables so we can easily test

**Comment by alfonsobries** [comment]
@patricio0312rev the table on one of the examples is not using the whole space,

image


Other than that 👌

**Review by alfonsobries — APPROVED** [review]

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
can we add just a table parser? there's no need to implement GFM for the articles as the other aspects aren't used.

Design will also be slightly different for tables than what is currently implemented. Turning this into draft until design is approved

File: resources/css/_article.css [hunk 2]
```diff
@@ -178,3 +178,39 @@
...
+.article-content table tbody td {
+    @apply p-4 text-base;
+}
+
+.article-content table tbody tr:not(:last-child) td {
+    @apply border-b border-dashed border-theme-secondary-300 text-theme-secondary-700 dark:border-theme-dark-700 dark:text-theme-dark-200;
+}
+
+.article-content table tbody tr:last-child td {
+    @apply pb-8;
```
Comment (ItsANameToo) [inline resources/css/_article.css:215]: this is a very large padding 🤔

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
some general remarks:
- we should discuss the need for GFM and how it's currently implemented in this PR if we choose to keep it
- some articles were adjusted to include a table (e.g. a TOC) but those link to sections while we don't support anchors, making them contain examples that don't work as intended. We should discuss the need for heading anchors and if needed implement them or adjust the examples to not set false expectations
- tables will need design adjustments, I'll link that on the clickup card

**Review by samharperpittam — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**